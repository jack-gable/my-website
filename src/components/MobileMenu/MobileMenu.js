"use client";
import React from "react";
import Link from "next/link";
import * as Dialog from "@radix-ui/react-dialog";
import VisuallyHidden from "../VisuallyHidden";
import { FiMenu as Menu, FiX as X } from "react-icons/fi";

import styles from "./MobileMenu.module.css";

function MobileMenu({ isOpen, setOpen, onDismiss }) {
	return (
		<Dialog.Root open={isOpen} onOpenChange={setOpen}>
			<Dialog.Trigger asChild>
				<div className={styles.menuBtn} role="button" aria-label="menu">
					<Menu size="1.5rem" />
					<VisuallyHidden>Open menu</VisuallyHidden>
				</div>
			</Dialog.Trigger>
			<Dialog.Portal>
				<Dialog.Overlay />
				<Dialog.Content className={styles.wrapper}>
					<div className={styles.backdrop}></div>
					<div className={styles.content}>
						<div className={styles.innerWrapper}>
							<Dialog.Close asChild>
								<div className={styles.closeBtn}>
									<X size="2.2rem" />
									<VisuallyHidden>
										Dismiss menu
									</VisuallyHidden>
								</div>
							</Dialog.Close>

							<nav className={styles.nav}>
								<ul className={styles.navList}>
									<li
										className={styles.listItem}
										onClick={onDismiss}
									>
										<Link
											className={styles.navLink}
											href="/about"
											aria-label="about"
										>
											<div>ABOUT</div>
										</Link>
									</li>
									<li
										className={styles.listItem}
										onClick={onDismiss}
									>
										<Link
											className={styles.navLink}
											href="/projects"
											aria-label="projects"
										>
											<div>PROJECTS</div>
										</Link>
									</li>
									<li
										className={styles.listItem}
										onClick={onDismiss}
									>
										<Link
											className={styles.navLink}
											href="/contact"
											aria-label="contact"
										>
											<div>CONTACT</div>
										</Link>
									</li>
									<li
										className={styles.listItem}
										onClick={onDismiss}
									>
										<Link
											className={styles.navLink}
											href="https://www.linkedin.com/in/jack-gable/"
											target="_blank"
											rel="noreferrer"
										>
											<div>LINKEDIN</div>
										</Link>
									</li>
									<li
										className={styles.listItem}
										onClick={onDismiss}
									>
										<Link
											className={styles.navLink}
											href="https://github.com/jack-gable"
											target="_blank"
											rel="noreferrer"
										>
											<div>GITHUB</div>
										</Link>
									</li>
								</ul>
							</nav>
						</div>
					</div>
				</Dialog.Content>
			</Dialog.Portal>
		</Dialog.Root>
	);
}

export default MobileMenu;
