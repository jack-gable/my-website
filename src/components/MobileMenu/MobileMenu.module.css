.wrapper {
  position: fixed;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  background: transparent;
  display: flex;
  justify-content: flex-end;
  z-index: 10;
}

.backdrop {
  position: absolute;
  top: 0;
  bottom: 0;
  left: 0;
  right: 0;
  background: var(--color-backdrop);
  opacity: 0.8;
  animation: fadeIn 500ms;
}

.content {
  --overfill: 16px;
  position: relative;
  background: linear-gradient(
    to right,
    var(--color-backdrop),
    var(--color-backdrop-highlight)
  );
  width: calc(450px + var(--overfill));
  height: 100%;
  margin-right: calc(var(--overfill) * -1);
  padding: 24px 32px;
}

.innerWrapper {
  display: flex;
  flex-direction: column;
  height: 100%;
  animation: fadeIn 600ms both;
  animation-delay: 400ms;
}

.closeBtn {
  background-color: none;
  border: none;
  position: absolute;
  right: 24px;
  padding: 16px;
}

.menuBtn {
  padding: 16px;
}

.nav {
  display: flex;
  justify-content: center;
  align-items: center;
  margin-top: 8rem;
}

.navList {
  display: flex;
  gap: 32px;
  padding: 0;
  list-style-type: none;
  flex-direction: column;
  align-items: center;
}

.listItem {
  position: relative;
}

.navLink {
  border: none;
  background: transparent;
  cursor: pointer;
  border-radius: 12px;
  display: flex;
  align-items: center;
  gap: 12px;
  position: relative;
  padding: 10px 16px;
  text-decoration: none;
  color: var(--color-text);
  font-weight: 600;
  font-size: 1.25rem;
  transition: background 200ms;
}

.navLink:hover,
.navLink:focus {
  background: var(--color-decorative-500);
}

@media (min-width: 35rem) {
  .menuBtn {
    display: none;
  }
}

@media (prefers-reduced-motion: no-preference) {
  .content {
    animation: slideIn 500ms both cubic-bezier(0, 0.6, 0.32, 1.06);
    animation-delay: 200ms;
  }
}

@keyframes fadeIn {
  from {
    opacity: 0;
  }
  to {
    opacity: 1;
  }
}

@keyframes slideIn {
  from {
    transform: translateX(100%);
  }
  to {
    transform: translateX(0%);
  }
}
