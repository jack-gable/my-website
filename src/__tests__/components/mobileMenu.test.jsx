import React from "react";
import MobileMenu from "@/components/MobileMenu";
import "@testing-library/jest-dom";
import { render, fireEvent } from "@testing-library/react";

describe("<MobileMenu />", () => {
	it("renders the <MobileMenu /> component with text", () => {
		const { container, getByText } = render(<MobileMenu />);

		fireEvent.click(getByText("Open menu"));
		expect(getByText("ABOUT")).toBeInTheDocument();
		expect(getByText("PROJECTS")).toBeInTheDocument();
		expect(getByText("CONTACT")).toBeInTheDocument();
		expect(getByText("LINKEDIN")).toBeInTheDocument();
		expect(getByText("GITHUB")).toBeInTheDocument();
		expect(container.firstChild).toMatchSnapshot();
	});
});
