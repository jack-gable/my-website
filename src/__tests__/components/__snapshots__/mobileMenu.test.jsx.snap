// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MobileMenu /> renders the <MobileMenu /> component with text 1`] = `
<div
  aria-controls="radix-:r0:"
  aria-expanded="true"
  aria-haspopup="dialog"
  aria-label="menu"
  class="menuBtn"
  data-state="open"
  role="button"
  type="button"
>
  <svg
    fill="none"
    height="1.5rem"
    stroke="currentColor"
    stroke-linecap="round"
    stroke-linejoin="round"
    stroke-width="2"
    viewBox="0 0 24 24"
    width="1.5rem"
    xmlns="http://www.w3.org/2000/svg"
  >
    <line
      x1="3"
      x2="21"
      y1="12"
      y2="12"
    />
    <line
      x1="3"
      x2="21"
      y1="6"
      y2="6"
    />
    <line
      x1="3"
      x2="21"
      y1="18"
      y2="18"
    />
  </svg>
  <div
    class="wrapper"
  >
    Open menu
  </div>
</div>
`;
